File: java11/src/main/java/com/chilvers/henry/acer/Player.java
package com.chilvers.henry.acer;

import lombok.*;

import java.util.List;
import java.util.stream.Collectors;

@Getter
@Setter
@AllArgsConstructor
@ToString
public class Player {
    private int position;
    private String name;
    private List<Integer> daysToChoose;

    public String finalResults() {
        return this.getDaysToChoose().stream()
         .sorted()
         .map(Object::toString)
         .collect(Collectors.joining(","));
    }
}

File: kotlin/src/main/kotlin/com/chilvers/henry/acer/Player.kt
package com.chilvers.henry.acer

data class Player(val position: Int, val name: String) {
    var daysToChoose: MutableList<Int> = mutableListOf()

    fun finalResults(): String {
        return daysToChoose
            .sorted()
            .joinToString("," )
    }
}